toucan_connectors/toucan_connector.py

```diff
@@ -183,3 +183,31 @@ def get_df_and_count(self, data_source: ToucanDataSource, limit: Union[int, None
     def explain(self, data_source: ToucanDataSource):
         """Method to give metrics about the query"""
         return None
+
+    @staticmethod
+    def check_hostname(hostname) -> bool:
+        """Check if a hostname is resolved"""
+        try:
+            socket.gethostbyname(hostname)
+            return True
+        except socket.error:
+            return False
+
+    @staticmethod
+    def check_port(host, port) -> bool:
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
```

Comment:
Don't you think we should catch any exception here, despite the usage of `connect_ex` ? AFAIK, you could still have exceptions, for instance if port is not a valid value (e.g. a port out of `unsigned short` range)?

Reply:
Done !